syntax="proto3";
package pms;

option optimize_for = LITE_RUNTIME;

// the sever logic module name
// it userd between servers, not used in client
enum ETransferModule{
    MCONNECTOR  = 0;
    MMSGQUEUE   = 1;
    MMEETING    = 2;
    MP2P        = 3;
    MLIVE       = 4;
    MCALLCENTER = 5;
    MLOGICAL    = 6;
    MSEQUENCE   = 7;
    MSTORAGE    = 8;
    MGRPNOTIFY  = 9;
    MPUSHER     = 10;
}

// the type of server transfering msg
enum ETransferType{
    TCONN         = 0;
    TTRANS        = 1;
    TQUEUE        = 2;
    TDISPATCH     = 3;
    TPUSH         = 4;
    TLOGIN        = 5;
    TLOGOUT       = 6;
    TWRITE_REQUEST      = 7;
    TWRITE_RESPONSE     = 8;
    TREAD_REQUEST       = 9;
    TREAD_RESPONSE      = 10;
    TERROR      = 11;
}

// the priority of transfering msg
// default is normal
enum ETransferPriority{
    PLOW        = 0;
    PNORMAL     = 1;
    PHIGH       = 2;
}

// just the flag
enum ETransferFlag{
    FNOACK      = 0;
    FNEEDACK    = 1;
    FACK        = 2;
}

enum ETransferTag{
    SEQN        = 0;
    DATA        = 1;
}

// the tag used for connecting of module
enum EConnTag{
    THI         = 0;
    THELLO      = 1;
    THELLOHI    = 2;
    TKEEPALIVE  = 3;
}

// the type of connection
enum EConnType{
    TTCP        = 0;
    THTTP       = 1;
}
